tests/cross/src/test/scala/tests/pc/CompletionExtensionMethodSuite.scala

```diff
@@ -427,41 +508,119 @@ class CompletionExtensionMethodSuite extends BaseCompletionSuite {
+       |object Test {
+       |  implicit class TestOps(val testArg: Int) {
        |    var testVar: Int = 42
        |    val testVal: Int = 42
        |    def testOps(b: Int): String = ???
+       |  }
+       |}
        |
-       |def main = 100.test@@
+       |object O {
+       |  def main = 100.test@@
+       |}
        |""".stripMargin,
     """|testArg: Int (implicit)
        |testVal: Int (implicit)
        |testVar: Int (implicit)
        |testOps(b: Int): String (implicit)
-       |""".stripMargin
+       |""".stripMargin,
+    compat = Map(
+      "2" ->
+        """|testArg: Int (implicit)
+           |testOps(b: Int): String (implicit)
+           |testVal: Int (implicit)
+           |testVar: Int (implicit)
+           |""".stripMargin
+    ),
+    filter = _.contains("(implicit)")
   )
 
   checkEdit(
     "implicit-val-edit".tag(IgnoreForScala3CompilerPC),
     """|package example
        |
-       |object Test:
-       |  implicit class TestOps(a: Int):
+       |object Test {
+       |  implicit class TestOps(a: Int) {
        |    val testVal: Int = 42
+       |  }
+       |}
        |
-       |def main = 100.test@@
+       |object O {
+       |  def main = 100.test@@
+       |}
        |""".stripMargin,
     """|package example
        |
        |import example.Test.TestOps
        |
-       |object Test:
-       |  implicit class TestOps(a: Int):
+       |object Test {
+       |  implicit class TestOps(a: Int) {
        |    val testVal: Int = 42
+       |  }
+       |}
        |
-       |def main = 100.testVal
-       |""".stripMargin
+       |object O {
+       |  def main = 100.testVal
+       |}
+       |""".stripMargin,
+    filter = _.contains("(implicit)")
+  )
+
+  check(
+    "complex-type-old",
+    """|package example
+       |
+       |object Test {
+       |  implicit class TestOps(a: List[Int]) {
+       |    def testOps(b: Int) = ???
+       |  }
+       |}
+       |
+       |object ActualTest {
+       |  List(1).tes@@
+       |}
+       |""".stripMargin,
+    "testOps(b: Int): Nothing (implicit)",
+    filter = _.contains("testOps")
+  )
+
+  check(
+    "complex-type-old2".tag(IgnoreScala3),
+    """|package example
+       |
+       |object Test {
+       |  implicit class TestOps[T](a: List[T]) {
+       |    def testOps(b: Int) = ???
+       |  }
+       |}
+       |
+       |object ActualTest {
+       |  List(1).tes@@
+       |}
+       |""".stripMargin,
+    "testOps(b: Int): Nothing (implicit)",
+    filter = _.contains("testOps")
+  )
+
+  check(
+    "complex-type-old3".tag(IgnoreScala3),
+    """|package example
+       |
+       |case class A[-T](t: T)
+       |
+       |object Test {
+       |  implicit class TestOps[T](a: A[T]) {
+       |    def testOps(b: Int) = ???
+       |  }
+       |}
+       |
+       |object ActualTest {
+       |  A(1).tes@@
+       |}
+       |""".stripMargin,
+    "testOps(b: Int): Nothing (implicit)",
+    filter = _.contains("testOps")
```

Comment:
To make sure there are no incorrect completions?
```suggestion
    filter = _.contains("implicit")
```